clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3171,18 +3209,49 @@ static void handleSubGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 // Handles num_simd_work_items.
-static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D, const ParsedAttr &A) {
+static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   if (D->isInvalidDecl())
     return;
 
-  Expr *E = A.getArgAsExpr(0);
+  Expr *E = AL.getArgAsExpr(0);
 
   if (D->getAttr<SYCLIntelNumSimdWorkItemsAttr>())
-    S.Diag(A.getLoc(), diag::warn_duplicate_attribute) << A;
+    S.Diag(AL.getLoc(), diag::warn_duplicate_attribute) << AL;
 
-  S.CheckDeprecatedSYCLAttributeSpelling(A);
+  S.CheckDeprecatedSYCLAttributeSpelling(AL);
+
+  if (!E->isValueDependent()) {
+    ASTContext &Ctx = S.getASTContext();
+    Optional<llvm::APSInt> ArgVal = E->getIntegerConstantExpr(Ctx);
+
+    if (const auto *A = D->getAttr<ReqdWorkGroupSizeAttr>()) {
+      Optional<llvm::APSInt> XDimVal = A->getXDimVal(Ctx);
+      Optional<llvm::APSInt> YDimVal = A->getYDimVal(Ctx);
+      Optional<llvm::APSInt> ZDimVal = A->getZDimVal(Ctx);
+
+      assert(ArgVal && XDimVal &&
+             "Argument should be an integer constant expression");
+      assert(ArgVal && YDimVal &&
+             "Argument should be an integer constant expression");
+      assert(ArgVal && ZDimVal &&
+             "Argument should be an integer constant expression");
+
+      int64_t NumSimdWorkItems = ArgVal->getSExtValue();
+      if (NumSimdWorkItems == 0)
```

Comment:
I think this situation is missing a diagnostic (we're silently dropping the attribute otherwise). I'd recommend hoisting this out of the innermost `if` block and diagnosing it there (and be sure to remove the extra checking for this case from `addIntelSYCLSingleArgFunctionAttr` as I think that code path should no longer be reachable).

Reply:
Done